OneCommunityGlobal/HighestGoodNetworkApp
# Papia added myteam filter and update user team

# Description
(PRIORITY High) Jae: Create “My Team” filter for Tasks Tab and Leaderboard (WIP Papia)
On the Dashboard, we need a “My Team” button and filter. Clicking this button should filter both the Tasks Tab and Leaderboard to show only the people on that person’s team. Then they should be able to use the 24h, 48h, etc. buttons and only see their team members. 
Make it the Default option for Core Team members
This is for Owner, Admin and Core Team classes to have a way to only see teams they are on in both the Leaderboard and Tasks Tab views (see below)
This is needed because these classes see ALL users and that makes useless all the tools for easily reviewing the work of just their team 

##Related PR:
 This PR is updated version of frontend PR #913.

## Main changes explained:
Updated the assign team and delete team function in the team page and profile page, so that the user teams state got updated. Added 'My Team' toggle button. Added a toggleTeamView function and getMyTeam helper function to get data for logged in user's team members. Remove fetching leaderBoardData from mapStateToProps to leaderboard component, as it is fetching the wrong state.


## How to test:
1. check into current branch
2. npm run start:local to run this PR locally
3. Log as admin, go to other links -> teams page, assign yourself on  team or delete from team, do the same on view profile-> team tab.
4. Go to dashboard, user will show team toggler button and their team members by default if they are assigned to any team, on leadderboard and task tab. If they are not assigned to any team, will not see the button and will see all members.
toggele the button to see if it shows all members and the person's team members alternatively. And also able to ftech correct data for clicking 24hr,48hr,72hr buttons. Also check the responsivenes.
4.Log in as volunteer or manager to see they can't see the filter button, and correct data of members are fetched on task and leaderboard 


## Screenshots or videos of changes:


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/86232710/d4548ef6-7621-423f-867e-412540bdf214

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/86232710/9b0135b4-5567-4b28-8839-413cb95313f6
